.github/workflows/ts_serialize_tests.yml

```diff
@@ -40,7 +40,7 @@ jobs:
         run: deno cache test_deps.ts
 
       - name: Deno test
-        run: deno test
+        run: deno test --doc
```

Comment:
Note that this doesn't actually run the tests in the READMEs, it only does type checking. I wrote a quick script to extract the tests and run them:
`deno run --allow-read --allow-write --allow-run run-doc-tests.ts README.md`
`run-doc-tests.ts`
```ts
import { basename } from "https://deno.land/std@0.127.0/path/mod.ts";

const args = Deno.args;

const testPrefix = [
  'import { test } from "./test_deps.ts";',
  "test({",
  "name:'test',",
  "fn() {",
];

const testPostfix = [
  "},",
  "});",
  "",
];

const docTestsPath = ".";

for (const arg of args) {
  await extractTests(arg);
}

async function extractTests(file: string) {
  const text = await Deno.readTextFile("./README.md");
  const textLines = text.split("\n");

  let testCounter = 0;
  let testLines: string[] | null = null;

  for (const line of textLines) {
    // Start of test
    if (testLines == null && line == "```ts") {
      testLines = [];
    } // End of test
    else if (testLines != null && line == "```") {
      // Create test file
      const outputPath = `${docTestsPath}/${
        basename(file)
      }.${testCounter}.test.ts`;

      const testImportLines = testLines.filter((line) =>
        line.startsWith("import")
      );
      const testBodyLines = testLines.filter((line) =>
        !line.startsWith("import")
      );
      const outputLines = testImportLines.concat(testPrefix).concat(
        testBodyLines,
      ).concat(testPostfix);
      const outputText = outputLines.join("\n");

      console.log(`Writing test to ${outputPath}`);

      await Deno.writeTextFile(outputPath, outputText);

      console.log(`Running tests...`);
      const process = Deno.run({ cmd: ["deno", "test", outputPath] });
      await process.status();

      console.log(`Removing tests...`);
      await Deno.remove(outputPath);

      testLines = null;
      testCounter++;
    } // Body of test
    else if (testLines != null) {
      testLines.push(line);
    }
  }
}
```

Reply:
I don't think the intent behind this PR was to run the READMEs as tests, so I think this level of functionality out of scope

Reply:
I made an update to add in the `test()` function, so I think you can reduce this script quite a bit now, thanks man!